modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
+///    PayInvoice -- await preimage offer accepted --> BuyPreimage
+///    PayInvoice -- await preimage offer failure --> Canceled
+///    BuyPreimage -- await federation decryption --> Preimage
+///    BuyPreimage -- external payment --> Preimage
+///    BuyPreimage -- await buy preimage failure --> Refund
+///    Preimage -- await claim outgoing contract --> Outpoint
```

Comment:
this is also not just "await", it triggers the transaction claiming the contract and then waits for its acceptance. Might deserve an intermediate state to properly model the failure condition of the tx being rejected for some reason.

Reply:
Yeah I've been kind of lumping the "transaction + additional action" into one transition. I'll split into two

Reply:
Hmm actually maybe I didn't understand this earlier. So this is actually "fetch outgoing contract and submit transaction claiming contract"?